include/picolibrary/microchip/megaavr/peripheral/spi.h

```diff
@@ -0,0 +1,487 @@
+     */
+    void configure_device() noexcept
+    {
+        spsr.configure_device();
+        spcr.configure_device();
+    }
+
+    /**
+     * \copydoc picolibrary::Microchip::megaAVR::Peripheral::SPI::SPCR::enable()
+     */
+    void enable() noexcept
+    {
+        spcr.enable();
+    }
+
+    /**
+     * \copydoc picolibrary::Microchip::megaAVR::Peripheral::SPI::SPCR::disable()
+     */
+    void disable() noexcept
+    {
+        spcr.disable();
+    }
+
+    /**
+     * \copydoc picolibrary::Microchip::megaAVR::Peripheral::SPI::SPCR::enable_interrupt()
+     */
+    void enable_interrupt() noexcept
+    {
+        spcr.enable_interrupt();
+    }
+
+    /**
+     * \copydoc picolibrary::Microchip::megaAVR::Peripheral::SPI::SPCR::disable_interrupt()
+     */
+    void disable_interrupt() noexcept
+    {
+        spcr.disable_interrupt();
+    }
+
+    /**
+     * \brief Configure SPI data exchange.
+     *
+     * \param[in] clock_rate The desired clock rate.
+     * \param[in] clock_polarity The desired clock polarity.
+     * \param[in] clock_phase The desired clock phase.
+     * \param[in] bit_order The desired data exchange bit order.
+     */
+    void configure( Clock_Rate clock_rate, Clock_Polarity clock_polarity, Clock_Phase clock_phase, Bit_Order bit_order ) noexcept
+    {
+        spsr.configure( clock_rate );
+        spcr.configure( clock_rate, clock_polarity, clock_phase, bit_order );
+    }
+
+    /**
+     * \copydoc picolibrary::Microchip::megaAVR::Peripheral::SPI::SPSR::exchange_complete()
+     */
+    auto exchange_complete() const noexcept
+    {
+        return spsr.exchange_complete();
+    }
+
+    /**
+     * \copydoc picolibrary::Microchip::megaAVR::Peripheral::SPI::SPSR::write_collision()
+     */
+    auto write_collision() const noexcept
+    {
+        return spsr.write_collision();
+    }
+
+    /**
+     * \brief Write data the transmit buffer.
+     *
+     * \param[in] data The data to write to the transmit buffer.
+     */
+    void transmit( std::uint8_t data ) noexcept
+    {
+        spdr = data;
+    }
+
+    /**
+     * \brief Read data from the receive buffer.
+     *
+     * \return The data read from the receive buffer.
+     */
+    auto receive() noexcept -> std::uint8_t
+    {
+        return spdr;
+    }
+};
+
+inline void SPI::SPCR::configure( Clock_Rate clock_rate, Clock_Polarity clock_polarity, Clock_Phase clock_phase, Bit_Order bit_order ) noexcept
+{
+    *this = ( *this & ~( Mask::DORD | Mask::CPOL | Mask::CPHA | Mask::SPR ) )
+            | ( ( static_cast<std::uint8_t>( clock_rate ) >> Offset::SPR ) & Mask::SPR )
+            | static_cast<std::uint8_t>( clock_polarity )
+            | static_cast<std::uint8_t>( clock_phase ) | static_cast<std::uint8_t>( bit_order );
+}
+
+inline void SPI::SPSR::configure( Clock_Rate clock_rate ) noexcept
+{
+    *this = ( *this & Mask::SPI2X ) | ( static_cast<std::uint8_t>( clock_rate ) & Mask::SPI2X );
```

Comment:
`SPI2X` is the only writable bit in the `SPSR` register. There is no reason to try to preserve the other bits. Having this function simply write to the register would improve performance.